File: src/main/java/io/gittul/domain/post/dto/NormalPostCreateRequest.java
package io.gittul.domain.post.dto;

import io.gittul.global.validation.OptionalURL;
import jakarta.validation.constraints.Size;

import java.util.List;

public record NormalPostCreateRequest(
        @Size(min = 1, max = 30)
        String title,

        @Size(min = 1, max = 200)
        String content,

        @OptionalURL
        String image,

        List<String> tags
) {
}

File: app/src/main/kotlin/io/gittul/app/domain/post/dto/NormalPostCreateRequest.kt
package io.gittul.app.domain.post.dto

import io.gittul.core.global.validation.OptionalURL
import jakarta.validation.constraints.Size

data class NormalPostCreateRequest(
    @field:Size(min = 1, max = 30)
    val title: String,

    @field:Size(min = 1, max = 200)
    val content: String,

    @field:OptionalURL
    val image: String?,

    val tags: List<String>
)
